### ISSUE TITLE
unable to download attachments from webapp 

### ISSUE THREAD

**Issue by tccadam** [moneymanagerex/moneymanagerex#2639]
Hi, 

I'm using MMEx Desktop version 1.3.5 (latest as of Jun 24 2020) on Windows.

This has happened a few times since I upgraded and the new "Import Webapp Transaction" window was implemented within the last couple of months.  I really like the new way the "Import Webapp Transaction" window appears automatically and checks the webapp for transactions to download, it works well, except this occasional problem.

Please see the screenshot below, I have had this error a few times, "unable to download attachments from webapp" If I click OK it will give me the error on the next transaction, so I have tried killing the mmex process and restarting, but it still gives me this error. I then have killed the process, rebooted my computer and reopened MMEx and that problem seems to stop happening for a few days or weeks.

![image](https://user-images.githubusercontent.com/64762525/85625283-7076c380-b628-11ea-8595-98f5cd5672c3.png)

Thanks,
Adam

**Comment by gabriele-v** [comment]
Hi, so if you press ok it will show the same message for the next transaction?
Does the next one has an attachment as well?

**Comment by tccadam** [comment]
yes. I got the same message for next transaction and yes, they all have attachments.
I forgot to mention that after clicking OK, I also get the error message that mmex was unable to insert the transaction into the database,

**Comment by mrcsmcl** [comment]
The same here.

Version: 1.3.5
• Database version: 7 • (aes128cbc)

Running on:
• Windows 10 (build 19041), 64-bit edition
• Portuguese (Brazilian) (windows-1252)
• 1920x1080 32bit 96x96ppi

**Comment by gabriele-v** [comment]
I'm not able to reproduce, are you using HTTPS or plain HTTP?
I'm releasing soon a new version with improved error messages in order to try sorting out the problem

**Comment by gabriele-v** [comment]
You should be able to download version with improved error messages here:
https://ci.appveyor.com/project/moneymanagerex/moneymanagerex/builds/33774356/job/83yng8aa4nx1pmdk/artifacts

**Comment by mrcsmcl** [comment]
I'm using plain HTTP, on my home server. The error only appears when downloading the attachment. I can upload on webapp from my phone/computer and it really shows on the server folder and mmex webapp screen.

**Comment by mrcsmcl** [comment]
Here's the messagens:

![image](https://user-images.githubusercontent.com/21992269/85927119-72d05c00-b87a-11ea-9ddf-1f1db05452c5.png)

![image](https://user-images.githubusercontent.com/21992269/85927125-7bc12d80-b87a-11ea-8cd7-0a6d4bee3fe2.png)

![image](https://user-images.githubusercontent.com/21992269/85927129-88de1c80-b87a-11ea-9a92-b490c4c275bf.png)

On the third image, what's in portuguese: "error 3: the system cannot find the specified path".

**Comment by gabriele-v** [comment]
@mrcsmcl looks to be something related to file save on disk, could be a race condition between file download and OneDrive?
Could you try do temporary disable or pause OneDrive, download and then re-enable it?

@tccadam are you saving attachments on OneDrive as well?

**Comment by mrcsmcl** [comment]
@gabriele-v It's not. I already tried using on desktop and It also gives an error. I'll teu again with this latest version you sent and post It here.

**Comment by tccadam** [comment]
Very interesting, I use Dropbox, but same idea. 

I never had this crash and error before the upgrade but it did use to take a really really long time to download and insert the webapp transactions sometimes. I mean like 15 minutes to import maybe 5 transactions which realistically would only take maybe 30 seconds to download from the webapp server. BUT I found that if I just waited and waited it would eventually finish so I lived with it.

**Comment by gabriele-v** [comment]
@tccadam could you try with version posted above and check if message it's the same?

**Comment by mrcsmcl** [comment]
It gives me the same error.

![image](https://user-images.githubusercontent.com/21992269/85931219-0d3f9800-b899-11ea-99a2-9ac8ec78b198.png)

![image](https://user-images.githubusercontent.com/21992269/85931220-1466a600-b899-11ea-9326-0e19f570e174.png)

![image](https://user-images.githubusercontent.com/21992269/85931223-1fb9d180-b899-11ea-9d50-7b8ab7efeef8.png)


I started using webapp after update, so I don't know how it is expected to work.

**Comment by tccadam** [comment]
Thanks, yes, I already installed the version you posted, but this problem isn't consistent and I don't even know how to replicate it so when it happens again I will post

**Comment by mrcsmcl** [comment]
My attachment is a .png image with 657kb, I don't know if that my help.

I would try the wait ideia from @tccadam, but the error shows up at the moment I press import all/import and open all.

**Comment by tccadam** [comment]
I think it must also be related to the database because I have 2 databases that I switch between frequently and the smaller database has always been able to import from webapp quickly. The larger database is the one that was super slow importing and now seems to crash sometimes. (I can't honestly remember if this has happened or not with the smaller database)

**Comment by mrcsmcl** [comment]
@tccadam mine is a new database with only 288kb. I tried with and older and bigger, but error still happens.

**Comment by tccadam** [comment]
I just uploaded 4 transactions with attachments and imported them to my larger database without any errors.  Sorry, it's hard to troubleshoot when it appears to happen randomly every few weeks for me, but I'll post again when it happens if you don't get it figured out today. Thanks!

**Comment by tccadam** [comment]
Ok, problem just happened again to me with the new version that provides more debugging info.  See screenshot below, extra messages says "timeout was reached"

![image](https://user-images.githubusercontent.com/64762525/86836455-c4d36780-c05a-11ea-81cf-b42b7e224065.png)

**Comment by tccadam** [comment]
Same thing happened again to me just now.  Maybe you could give me a version with a much longer timeout time and I could see if it stops.  Like I posted before, in the older version, some times I'd need to wait 15 mins but eventually all my transactions would download from webapp and appear in my mmex db.

**Comment by grumpy235** [comment]
@gabriele-v
Could this be a CURL issue ?
Curious to know why MMEX 1.3.5RC2 is delivered with "libcurl/7.59.0-DEV WinSSL" which was released March 2018 but the current version of CURL is 7.71.0 ?

**Comment by gabriele-v** [comment]
It could be related to LibCURL, because if it's something on MMEX code side I guess it will be show each time, instead it looks to be random (probably when network it's more unstable?).
I could try to build it with latest CURL version and post the EXE here so you can check if it's working better.

**Comment by gabriele-v** [comment]
[MMEX_CURL_7.71.1.zip](https://github.com/moneymanagerex/moneymanagerex/files/4952223/MMEX_CURL_7.71.1.zip)
@mrcsmcl @tccadam @grumpy235 Please try this version, 64Bit Windows with CURL 7.71.1

**Comment by mrcsmcl** [comment]
Thank you @gabriele-v, but it gives me the same error. I can even download the attachment separetelly, but when I try to import it fails.

**Comment by vomikan** [comment]
I think it should be some test  available for curl command line.
```
curl http://some.url --output some.file
```

**Comment by tccadam** [comment]
2 things to report:

1. I noticed a timeout setting in the MMEx network options, it was set to 10 seconds and I increased it to 150 (the max) and I had no idea that option existed, I assumed the timeout was in the code.

2. I just uploaded 7 transactions in the webapp. I opened MMEX and I clicked "import all" and it crashed on the first 6, and I clicked OK, OK, to the error messages each time they came up.  I watched the attachments appear in my dropbox folder and the date modified would first appear current, but then it would change to 3 hours ago.  Then after I clicked OK to the error message in MMEx the attachments would disappear from the dropbox folder.  The 7th and last attachment and transaction DID stay in MMEx and in the dropbox folder (with the incorrect time of 3 hours ago).  So I started to wonder if the server where the webapp is hosted has the wrong time, so I logged into the hosting provider and they didn't have any setting for date/time but I updated my profile and maybe that fixed the date/time also, but not sure.

Anyways, I just tested again with 3 transactions and they downloaded fine and the attachments have the correct time in dropbox. SO, I don't know if 1 or both of the things I did above helped or if this will happen again in a few weeks, but I thought I should share.
